pennylane/pulse/parametrized_hamiltonian.py

```diff
@@ -240,7 +240,18 @@ def __call__(self, params, t):
         return self.H_fixed() + self.H_parametrized(params, t)
 
     def __repr__(self):
-        return f"ParametrizedHamiltonian: terms={qml.math.shape(self.coeffs)[0]}"
+        terms = []
+        for i, (coeff, op) in enumerate(
+            zip(
+                self.coeffs_fixed + self.coeffs_parametrized, self.ops_fixed + self.ops_parametrized
+            )
+        ):
+            if callable(coeff):
+                term = f"({coeff.__name__}(params_{i}, t)*({op}))"
```

Comment:
We have to be a bit more careful here because also class objects are `callable`. The following example would raise an error:

```python
import pennylane as qml

def f1(p, t):
    return p[0] * jnp.sin(p[1] * t)

def f2(p, t):
    return p * t

class f3:
    def __init__(self, hyper_param):
        self.hyper = hyper_param
    def __call__(self, p, t):
        return self.hyper * p * t

coeffs = [2., f1, f2, f3(0.5)]
observables =  [qml.PauliX(0), qml.PauliY(0), qml.PauliZ(0), qml.PauliX(0)]
H = qml.dot(coeffs, observables)

print(H)
```

```pycon
AttributeError: 'f3' object has no attribute '__name__'
```

Reply:
I think a simple ```hasattr(coeff, "__name__")```
Should fix this problem